web/src/utils/uid.ts

```diff
@@ -0,0 +1,74 @@
+import { customAlphabet } from 'nanoid'
+
+const idGenerator: Record<number, () => string> = {}
+
+/**
+ * A random id generator using custom alphabet and length.
+ * Because instantiating the function can be expensive, we cache the generators
+ * based on the size of the ID required.
+ */
+const nanoid = (size = 20): string => {
+  let uidGenerator = idGenerator[size]
+
+  if (uidGenerator) {
+    return uidGenerator()
+  } else {
+    uidGenerator = customAlphabet(
+      '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz',
+      size
+    )
+
+    idGenerator[size] = uidGenerator
+    return uidGenerator()
+  }
+}
+
+/**
+ * Generate a unique identifier.
+ *
+ * Generated identifiers:
+ *
+ * - are URL safe
+ * - contain information on the type of object that they
+ *   are identifying
+ * - have an extremely low probability of collision
+ *
+ * Generated identifiers have a fixed length of 32 characters made up
+ * of three parts separated by dots:
+ *
+ * - 2 characters in the range `[a-z]` that identifying the "family" of
+ *   identifiers, usually the type of object
+ *   the identifier is for e.g. `rq` = request
```

Comment:
@nokome These docs are copy pasted, and state that `request` is abbreviated as `rq`, but in [Rust we use `re`](https://github.com/stencila/stencila/blob/117367ccc72abaffa8e2b45a22babed0cc5e8567/rust/node-execute/src/messages.rs#L5).
Shall I update the docs to `re`?

Reply:
Yes please